### PR TITLE
add gitignore file

### PR DESCRIPTION
**Pull request by wackxu**
/assign @apelisse

### CODE REVIEW

**Comment by apelisse** [comment]
I don't know about any of these files. If your OS/tools are leaving traces everywhere, please refer to [gitignore documentation](https://git-scm.com/docs/gitignore#_description), I think this belongs to `$GIT_DIR/info/exclude` or `$HOME/.config/git/ignore` (depending on `core.excludesFile` for the later)

**Comment by mengqiy** [comment]
@apelisse I have a different opinion of this PR.
I think we should do something similar as https://github.com/kubernetes/kubernetes/blob/master/.gitignore. IMHO project-based gitignore is more convenient than the user-based one.

**Comment by apelisse** [comment]
yeah, if that's what Kubernetes does :-)
Everyone has it's own IDE, and I don't think we should update the gitignore anytime somebody starts using a new one. But sure.

**Comment by mengqiy** [comment]
@wackxu IMO, gitignore can have at least the following. Can you update the PR?

```
# OSX leaves these everywhere on SMB shares
._*

# OSX trash
.DS_Store

# Eclipse files
.classpath
.project
.settings/**

# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
.idea/
*.iml

# Vscode files
.vscode

# This is where the result of the go build goes
/output*/
/_output*/
/_output

# Emacs save files
*~
\#*\#
.\#*

# Vim-related files
[._]*.s[a-w][a-z]
[._]s[a-w][a-z]
*.un~
Session.vim
.netrwhist

# cscope-related files
cscope.*

# Mercurial files
**/.hg
**/.hg*
```

**Comment by wackxu** [comment]
@mengqiy Updated

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+*.un~
+Session.vim
+.netrwhist
+
+# cscope-related files
+cscope.*
+
+# Mercurial files
+**/.hg
+**/.hg*
```
Comment (mengqiy) [inline .gitignore]: We'd better have a newline here.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+.settings/**
+
+# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
+.idea/
+*.iml
+
+# Vscode files
+.vscode
+
+# This is where the result of the go build goes
```
Comment (mengqiy) [inline .gitignore]: These 4 lines don't apply to this repo ATM. Please remove.

**Pull request closed without merging**